bedrock/base/urlresolvers.py

```diff
@@ -38,6 +38,23 @@ def reverse(viewname, urlconf=None, args=None, kwargs=None, prefix=None):
 reverse_lazy = lazy(reverse, str)
 
 
+def get_language_map():
+    """
+    Return a complete dict of language -> URL mappings, including the canonical
+    short locale maps (e.g. es -> es-ES and en -> en-US).
+    :return: dict
+    """
+    LUM = settings.LANGUAGE_URL_MAP
+    langs = dict(LUM.items() + settings.CANONICAL_LOCALES.items())
+    # Add missing short locales to the list. This will automatically map
+    # en to en-GB (not en-US), es to es-AR (not es-ES), etc. in alphabetical
+    # order. To override this behavior, explicitly define a preferred locale
+    # map with the CANONICAL_LOCALES setting.
+    langs.update((k.split('-')[0], v) for k, v in LUM.items() if
+                 k.split('-')[0] not in langs)
+    return langs
```

Comment:
Does this need to be a function, or would it make more sense to construct this dict once, on module import, since it only varies with the settings are changed?